### PR TITLE
ibus: fix dconf db installation

### PR DESCRIPTION
**Pull request by symphorien**
###### Motivation for this change

Fix a warning when running ibus-daemon:

```
(ibus-dconf:15691): dconf-WARNING **: 21:49:24.018: unable to open file '/etc/dconf/db/ibus': Failed to open file ?/etc/dconf/db/ibus?: open() failed: No such file or directory; expect degraded performance
```

cc @jtojnar @worldofpeace about dconf.

###### Things done

* Tested that ibus still works and that the warning is gone in a vm
* ran the gnome and gnome-xorg tests to ensure the dconf.profiles options has not regressed

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/programs/dconf.nix [hunk 1]
```diff
@@ -22,18 +33,22 @@ in
       profiles = mkOption {
         type = types.attrsOf types.path;
         default = {};
-        description = "Set of dconf profile files.";
+        description = "Set of dconf profile files, installed at <literal>/etc/dconf/profiles/name<literal>.";
```
Comment (jtojnar) [inline nixos/modules/programs/dconf.nix:36]:
```suggestion
        description = "Set of dconf profile files, installed at <filename>/etc/dconf/profiles/<replaceable>name</replaceable><filename>.";
```

File: nixos/modules/programs/dconf.nix [hunk 2]
```diff
@@ -22,18 +33,22 @@ in
...
         internal = true;
       };
 
+      packages = mkOption {
+        type = types.listOf types.package;
+        default = [];
+        description = "A list of packages which provide dconf profiles and databases in <literal>$out/etc/dconf</literal>.";
```
Comment (jtojnar) [inline nixos/modules/programs/dconf.nix:43]: Not always the out output.

```suggestion
        description = "A list of packages which provide dconf profiles and databases in <filename>/etc/dconf</filename>.";
```

File: pkgs/tools/inputmethods/ibus/default.nix [hunk 3]
```diff
@@ -47,6 +48,14 @@ let
       makeWrapper ${glib.dev}/bin/glib-mkenums $out/bin/glib-mkenums --unset PYTHONPATH
     '';
   };
+  # make-dconf-override-db.sh needs to execute dbus-launch in the sandbox,
```
Comment (jtojnar) [inline pkgs/tools/inputmethods/ibus/default.nix:51]: What error does this cause?

**Comment by symphorien** [comment]
Thanks for your review comments. I incorporated them.

File: pkgs/tools/inputmethods/ibus/default.nix (hunk 3 above)
Comment (symphorien) [reply pkgs/tools/inputmethods/ibus/default.nix:51]:
```
dbus[19730]: Failed to start message bus: Failed to open "/etc/dbus-1/session.conf": No such file or directory
```

I wrote it in the comment.

**Comment by ttuegel** [comment]
The ibus aspect of this looks okay to me, but I don't know enough about the dconf aspect to say.

File: pkgs/tools/inputmethods/ibus/default.nix (hunk 3 above)
Comment (jtojnar) [reply pkgs/tools/inputmethods/ibus/default.nix:51]: Oh, right. I did not notice we were previously replacing the file with a blank script.

Comment (jtojnar) [reply pkgs/tools/inputmethods/ibus/default.nix:51]: Can confirm this works to copy the default value from GSettings schemas to the dconf database. Though I do not understand why would they do it, unless they do not use GSettings and access dconf database directly.

**Review by jtojnar — APPROVED** [review]

**Review by worldofpeace — APPROVED** [review]

**Comment by symphorien** [comment]
The ibus test does not pass, but it didn't pass before either:
before
```
# Start of ibus tests
# MESSAGE: Failed to generate IBusConfig: Configuration daemon is not running.
Bail out! ERROR:ibus-config.c:23:finish_create_config_async_success: assertion failed: (IBUS_IS_CONFIG (config))
FAIL: ibus/ibus-config.test (Child process killed by signal 6)

```
After:
```
# Start of ibus tests
ok 1 /ibus/create-config-async
Bail out! IBUS-FATAL-WARNING: org.freedesktop.IBus.Config.GetValue: GDBus.Error:org.freedesktop.DBus.Error.Failed: Config value [test:v1] does not exist.
FAIL: ibus/ibus-config.test (Child process killed by signal 5)
```
which looks better but not good enough yet...

**Comment by symphorien** [comment]
I fixed the test by running ibus-daemon inside the session via systemd instead of via ssh. It's slightly mysterious to me though.

**Review by worldofpeace — APPROVED** [review]

**Comment by worldofpeace** [comment]
> I fixed the test by running ibus-daemon inside the session via systemd instead of via ssh. It's slightly mysterious to me though.

It seems fine to me, and it works :sparkle:

**Comment by worldofpeace** [comment]
I noticed an issue with the tests that introduced an eval issue https://github.com/NixOS/nixpkgs/pull/85970#issue-408791040. Reverted for now.

**Comment by symphorien** [comment]
reopened and fixed in #85992

**Pull request merged**